### ISSUE TITLE
Firefox version planned?

### ISSUE THREAD

**Issue by yoshimo** [fpmon/fingerprinting-monitor#6]
Webextensions are usually compatible across browsers.
Firefox doesn't use unpacked extensions like Chrome though.
Do you have plans to create a firefox package of this addon as well?

**Comment by fpmon** [comment]
Hi, you can use FPMON with firefox in the following way:

1. Call this URL: _about:debugging#/runtime/this-firefox_
2. Click "Load temporary add-on"
3. Select the manifest.json of the FPMON folder

For some pages, the CORS mechanism might impair your results. 
Anyhow, we don't really work with Firefox and we emphasize using Chromium instead.
https://www.chromium.org/getting-involved/download-chromium

Thanks for your request :)

**Comment by Mis012** [comment]
seems to work well enough, thanks

---

> Anyhow, we don't really work with Firefox and we emphasize using Chromium instead.

considering the ever shrinking firefox usebase, I think at this point a very high percentage of firefox users use it precisely to hold off google becoming the next IE... so I don't think you will have any luck converting them to anything that is for <del>all intents and purposes</del> that intent and purpose chrome

**Comment by Stoiker** [comment]
> For some pages, the CORS mechanism might impair your results.

Can you elaborate on this? I don't see why only Firefox should be affected.

**Comment by fpmon** [comment]
For example, the results for breitbart.com are very different because some resources are not loaded.
`Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://hb.emxdgt.com/?t=3000&ts=1609793548395&src=pbjs. (Reason: CORS request did not succeed)`

This does not happen when using chromium, etc.

**Comment by 3ventic** [comment]
> `Cross-Origin Request Blocked: The Same Origin Policy disallows reading the remote resource at https://hb.emxdgt.com/?t=3000&ts=1609793548395&src=pbjs. (Reason: CORS request did not succeed)`

That is due to Firefox's tracking protection being enabled and is an indication that it's working as intended. In the network tab you can see the failure reason is `Tracking`.

**Comment by fpmon** [comment]
Exactly the same issues occur when fingerprinting protection is disabled/enabled.
Same for tracking content option disabled/enabled.

Unfortunately, Mozilla is not (really) responding to our request since early November.
Since they are not interested to improve their solution, we don't want to put more effort into it.

**Comment by 3ventic** [comment]
> Exactly the same issues occur when fingerprinting protection is disabled/enabled.
> Same for tracking content option disabled/enabled.

Works here, so I'm not sure what the problem on your end is.

![image](https://user-images.githubusercontent.com/3998628/104432011-db69f680-5588-11eb-8c9e-fb01bd714e55.png)

**Comment by fpmon** [comment]
Ok, then everything seems to work as intended! Thanks 😊

**Comment by ke352802081770314** [comment]
> Unfortunately, Mozilla is not (really) responding to our request since early November.
> Since they are not interested to improve their solution, we don't want to put more effort into it.

Indeed, Mozilla bug handling can be quite erratic. I found years-old bugs that bit me which are just left there to rot, it seems. Do you happen to have the bugzilla bug number you opened so that readers may try to push it, or at least better understand the issue and possible impact?

If you define the objective of the add-on as "help the user see their exposure to fingerprinting" rather than "research the attempts of a site to perform fingerprinting, whether it succeeds or not" then benefitting from browser "fingerprinting support deficiencies" (whether intended or not by the developers and the user)  seems perfectly okay to me.
Or maybe I misunderstood the issue with Firefox, but I am happy to read your request to them and understand more from that, so you need not elaborate here.

I appreciate your instructions above, but if I interpret "Load temporary add-on" correctly, it may have to be repeated after each browser restart?
In any case, I would appreciate if you could create a release of this addon as a xpi file to make it easier to install in Firefox. That would make your work more accessible to more users. Many thanks in advance for considering this again.

**Comment by br4nnigan** [comment]
> Hi, you can use FPMON with firefox in the following way:
> 
>     1. Call this URL: _about:debugging#/runtime/this-firefox_
> 
>     2. Click "Load temporary add-on"
> 
>     3. Select the manifest.json of the FPMON folder
> 
> 
> For some pages, the CORS mechanism might impair your results. Anyhow, we don't really work with Firefox and we emphasize using Chromium instead. https://www.chromium.org/getting-involved/download-chromium
> 
> Thanks for your request :)

There's another approach: 

1. Use [Waterfox](waterfox.net/), FirefoxDeveloperEdition or Firefox Nightly
2. allow unsigned addons (`xpinstall.signatures.required` in about:config)
3. add the following code to FPMON_extention/manifest.json (dont mess up the json structure):
    `"applications": { "gecko": { "id": "someuniqueid@example.com" } }`
4. zip the contents of FPMON_extention
5. in about:addons click on the gear wheel and "install addon from file"
6. voilá, you can use the addon permanently

it seems to work but apparently there might be issues, like missing addon icon
